### ISSUE TITLE
Inline markdown preview

### ISSUE THREAD

**Issue by fer0n** [aeharding/voyager#618]
With the new markdown preview I was wondering if an "inline" preview like this would make sense in the default text input field: 

![image](https://github.com/aeharding/voyager/assets/20093825/a8dcdbf4-137d-4de1-9ef0-48a46014099c)

That's the best of both worlds imo

**Comment by aeharding** [comment]
I agree. However there is no good open source library that does this. See: https://github.com/aeharding/voyager/issues/367#issuecomment-1666625236

The closest I saw was https://github.com/davidmyersdev/ink-mde which is still waaaay too bloated.

I'd like a lib which is:

 1. Lightweight (no code syntax highlighting, no external dependencies other than maybe an AST parser like [mdast](https://github.com/syntax-tree/mdast))
 2. Styles existing markdown code - essentially just adding styles to plaintext
 3. Is a drop-in replacement for a `<textarea>`
 4. Offers lightweight, helpful completion for lists

Converting to a discussion since there no open source library currently exists.
